### DISCUSSION TITLE
Error with Module Swiper from Breeze

### DISCUSSION THREAD

**Discussion by leonardotbb** [breezefront/community#253] [1. Help] [answered]
Hello,

Since the last update of Breeze, the Module Swiper initialized from <theme_dir>/web/js/breeze/custom.js are not working. The error is "Swiper is not defined"

If you initialize them directly in the HTML, like this, they work,
        <div class="swiper-container swiper-about" 
            data-mage-init='{"Swissup_Swiper/js/swiper" : 
                {
                "slidesPerView": "auto",
            }'
        >

 but if you need to initialize them from a JavaScript file <theme_dir>/web/js/breeze/custom.js, it no longer works. 

            const mySwiper = new Swiper('.block-workout .swiper-pills', {
                spaceBetween: 20,
                slidesPerView: 1.7,
        }

This issue arises from a piece of code added in the latest update in the following file: /vendor/swissup/module-swiper/view/frontend/layout/breeze_default.xml.

                                <item name="load" xsi:type="array">
                                    <item name="onReveal" xsi:type="array">
                                        <item name="0" xsi:type="string">[data-mage-init*="Swissup_Swiper/js/swiper"]</item>
                                    </item>
                                </item>

How can we now initialize a Swiper from a <theme_dir>/web/js/breeze/custom.js file?

Thank you.

**Comment by vovayatsyuk** [comment, marked as answer]
Hello,

The error was caused by our "dynamic js" feature - since Swiper is a pretty big library we load it on demand.

Here is how you can use it from custom.js file:

```js
require(['Swissup_Swiper/js/swiper'], (swiperWidget) => {
    // swiperWidget - is a constructor function. Example: swiperWidget(settings, element);
    // Or create Swiper without widget using: new Swiper()
})
```

Since you are not using Swiper across all pages I recommend wrapping this code into "onReveal" callback:

```js
$.async('.block-workout .swiper-pills', (el) => {
    $.onReveal(el, () => {
        require(['Swissup_Swiper/js/swiper'], (swiperWidget) => {
            // swiperWidget - is a constructor function. 
            swiperWidget(settings, element); 
            // Or: $(el).swiper(settings);
        });

        // Or: create Swiper without widget usage: 
        require(['Swissup_Swiper/lib/swiper'], () => {
            new Swiper();
        });
    });
});
```

p.s. Updated with `Swissup_Swiper/lib/swiper` example.

**Comment by leonardotbb** [comment]
Thnx @vovayatsyuk . It works
